src/infill/SierpinskiFill.cpp

```diff
@@ -749,8 +749,49 @@ Polygon SierpinskiFill::generateCross(coord_t z, coord_t min_dist_to_side) const
     }
     assert(last_triangle);
     ret.add(get_edge_crossing_location(last_triangle->depth, last_triangle->getToEdge()));
-    
-    return ret;
+
+    if (pocket_size > 10)
+    {
+        coord_t sqrt_pocket_size = pocket_size * sqrt2 / 2;
+        std::cerr << sqrt_pocket_size << '\n';
+        
+        Polygon pocketed;
+        //pocketed.reserve(ret.size() * 3 / 2);
+
+        Point p0 = ret.back();
+        for (size_t poly_idx = 0; poly_idx < ret.size(); poly_idx++)
+        {
+            Point p1 = ret[poly_idx];
+            Point p2 = ret[(poly_idx + 1) % ret.size()];
+            Point v0 = p0 - p1;
+            Point v1 = p2 - p1;
```

Comment:
These are oriented in reverse. Was that intended? I'd expect either:

    Point v0 = p1 - p0;
    Point v1 = p2 - p1;

or:

    Point v0 = p0 - p1;
    Point v1 = p1 - p2;